compiler/src/dotty/tools/dotc/core/Mode.scala

```diff
@@ -122,6 +122,9 @@ object Mode {
 
   val PatternOrTypeBits: Mode = Pattern | Type
 
+  /** Skip inlining of unapply methods. */
+  val NoInlineUnapply: Mode = newMode(18, "NoInlineUnapply")
```

Comment:
Note that this is the last free mode. If we want more modes we will need to use a Long to encode them.